metron-platform/metron-parsers/src/test/java/org/apache/metron/writers/integration/WriterBoltIntegrationTest.java

```diff
@@ -166,6 +185,139 @@ public void test() throws UnableToStartException, IOException, ParseException {
     }
   }
 
+  /**
+   * {
+   *    "parserClassName":"org.apache.metron.writers.integration.WriterBoltIntegrationTest$EmptyObjectParser",
+   *    "sensorTopic":"emptyobjectparser",
+   *    "outputTopic": "enrichments",
+   *    "errorTopic": "parser_error"
+   * }
+   */
+  @Multiline
+  public static String offsetParserConfigJSON;
+
+  @Test
+  public void commits_kafka_offsets_for_emtpy_objects() throws Exception {
+    final String sensorType = "emptyobjectparser";
+
+    SensorParserConfig parserConfig = JSONUtils.INSTANCE.load(offsetParserConfigJSON, SensorParserConfig.class);
+
+    // the input messages to parser
+    final List<byte[]> inputMessages = new ArrayList<byte[]>() {{
+      add(Bytes.toBytes("foo"));
+      add(Bytes.toBytes("bar"));
+      add(Bytes.toBytes("baz"));
+    }};
+
+    // setup external components; zookeeper, kafka
+    final Properties topologyProperties = new Properties();
```

Comment:
Both test cases **seem** to do the same or similar component setup for Zk, Kafka, etc.  Can we extract any duplicate logic into a shared method?  Maybe just creating a "setup(@Before)" and "teardown(@After)" methods could handle that logic.

Pulling that logic out will also make it easier for me to understand the specifics of what each test case is intended to test.

Reply:
I'll look at pulling common logic.